tests/test_text_to_num_ru.py

```diff
@@ -0,0 +1,217 @@
+
+        source = "Сам по себе я одиночка"
+        self.assertEqual(source, alpha2digit(source, 'ru'))
+
+    def test_and(self):
+        source = "пятьдесят, шестьдесят, тридцать и одиннадцать"
+        expected = "50, 60, 30 и 11"
+        self.assertEqual(expected, alpha2digit(source, 'ru'))
+
+    def test_alpha2digit_zero(self):
+        source = "тринадцать тысяч, ноль девяносто"
+        expected = "13000, 090"
+        self.assertEqual(expected, alpha2digit(source, 'ru'))
+
+        self.assertEqual("0", alpha2digit("ноль", 'ru'))
+
+    def test_alpha2digit_ordinals_force(self):
+        source = (
+            "Пятый, третий, второй, двадцать первый, сотый, тысяча двести тридцатый, двадцать пятый, тридцать восьмой, сорок девятый."
+        )
+        expected = "5ый, 3ий, 2ой, 21ый, 100ый, 1230ый, 25ый, 38ой, 49ый."
+        self.assertEqual(expected, alpha2digit(source, 'ru', ordinal_threshold=0))
+        source = (
+            "первый, второй, третий, четвёртый, четвертый, пятый, шестой, седьмой, восьмой, девятый, десятый."
+        )
+        expected = "1ый, 2ой, 3ий, 4ый, 4ый, 5ый, 6ой, 7ой, 8ой, 9ый, 10ый."
+        self.assertEqual(expected, alpha2digit(source, 'ru', ordinal_threshold=0))
+
+        source = "двадцать второе место на двадцать первой олимпиаде занял первый и второй"
+        expected = "22ое место на 21ой олимпиаде занял 1ый и 2ой"
+        self.assertEqual(expected, alpha2digit(source, 'ru', ordinal_threshold=0))
+
+        source = "каждый пятый на первый второй расчитайсь!"
+        expected = "каждый 5ый на 1ый 2ой расчитайсь!"
+        self.assertEqual(expected, alpha2digit(source, 'ru', ordinal_threshold=0))
+
+    def test_alpha2digit_decimals(self):
+        source = (
+            "двенадцать точка девяносто девять, сто двадцать точка ноль пять,"
+            " сто двадцать целых ноль пять, одна целая двести тридцать шесть."
+        )
+        expected = "12.99, 120.05, 120.05, 1.236."
+        self.assertEqual(expected, alpha2digit(source, 'ru'))
+
+        self.assertEqual("0.15", alpha2digit("точка пятьнадцать", 'ru'))
+        self.assertEqual("0.15", alpha2digit("ноль целых пятьнадцать", 'ru'))
+
+    def test_alpha2digit_signed(self):
+        source = "В комнате плюс двадцать градусов, тогда как на улице минус пятьдесят."
+        expected = "В комнате +20 градусов, тогда как на улице -50."
+        self.assertEqual(expected, alpha2digit(source, 'ru'))
+
+    def test_uppercase(self):
+        source = "ПЯТЬНАДЦАТЬ, ОДИН, ДЕСЯТЬ, ОДИН"
+        expected = "15, 1, 10, 1"
+        self.assertEqual(expected, alpha2digit(source, 'ru'))
+
+    def test_examples(self):
+        source = "пятьдесят один миллион пятьсот семьдесят восемь тысяч триста два"
+        expected = 51578302
+        self.assertEqual(expected, text2num(source, 'ru'))
+
+        source = "восемьдесят один"
+        expected = 81
+        self.assertEqual(expected, text2num(source, 'ru'))
+
+    def test_sets_of_ones(self):
```

Comment:
Can you explain this test please?

Reply:
it is test for all cases for hundred, thousand, million etc with similar letters and words
may be this is an extra test now, but it was helpful on the one of first steps of developing

Reply:
fixed